python/lsst/ctrl/bps/parsl/sites/work_queue.py

```diff
@@ -0,0 +1,140 @@
+            Parsl execution provider, e.g., SlurmProvider.
+        port : `int`, optional
+            Port used by work_queue module.  Default: 9000
+        worker_options : `str`, optional
+            Extra options to pass to work_queue workers, e.g., `"--memory=90000"`.
+            Default: `""`
+        wq_max_retries : `int`, optional
+            Number of retries for work_queue to attempt per job.  Set to `None`
+            to have it try indefinitely; set to `1` to have Parsl control the
+            retries.  Default: `1`
+        """
+        port = get_bps_config_value(self.site, "port", int, port)
+        worker_options = get_bps_config_value(self.site, "worker_options", str,
+                                              worker_options)
+        max_retries = get_bps_config_value(self.site, "wq_max_retries", int,
+                                           wq_max_retries)
+        return WorkQueueExecutor(
+            label=label,
+            provider=provider,
+            port=port,
+            worker_options=worker_options,
+            max_retries=max_retries,
+            shared_fs=True,
+            autolabel=False
+        )
+
+
+class SlurmWorkQueue(WorkQueue):
+    """Configuration for using Slurm-based WorkQueueExecutor.
+
+    This can be used directly as the site configuration where Slurm
+    is available.  For running on a single node, e.g., a laptop, a
+    `SingleNodeLauncher` is used, and Slurm need not be available.
+
+    The following BPS configuration parameters are recognized, overriding the
+    defaults:
+
+    - ``port`` (`int`): The port used by the work_queue module. Default: `9000`.
+    - ``worker_options (`str`): Extra options to pass to work_queue workers.
+      a typical option sets the memory available per worker, e.g.,
+      `"--memory=90000"`, which sets the available memory to 90 GB.
+      Default: `""`
+    - ``wq_max_retries`` (`int`): The number of retries that the work_queue
+      module will make when a task fails.  Set to `None` to have work_queue
+      retry forever; set to `1` to have retries managed by Parsl.  Default: 1
+    - ``nodes_per_block`` (`int`): The number of allocated nodes. Default: 1
+    """
+
+    def get_executors(self) -> List[ParslExecutor]:
+        """Get a list of executors to be used in processing
+        """
+        nodes_per_block = get_bps_config_value(self.site, "nodes_per_block",
+                                               int, 1)
+        provider_options = dict(nodes_per_block=nodes_per_block,
+                                init_blocks=0,
+                                min_blocks=0,
+                                max_blocks=1,
+                                parallelism=0,
+                                cmd_timeout=300)
+        if nodes_per_block > 1:
+            provider_options["launcher"] \
+                = SrunLauncher(overrides="-K0 -k --slurmd-debug=verbose")
+        provider = LocalProvider(**provider_options)
```

Comment:
A `LocalProvider`? I thought this class was for running Slurm? In fact, I don't see anything Slurm-related here, except the `SrunLauncher`, and that can't be enough to provide everything for a Slurm submission.

Reply:
Do you mean it's for when you're already inside a Slurm allocation, running on a cluster node?

Reply:
yes.

Reply:
Since this is a general-purpose class that others might investigate to use for their own site, I think the naming is important. How about `SrunWorkQueue`? The distinction being that it's using `srun` to launch workers, as opposed to using Slurm from the head node. What do you think?

Reply:
That sounds fine.  `SrunWorkQueue` it is.